src/core/Akka.TestKit/TestKitBase_AwaitConditions.cs

```diff
@@ -347,43 +258,24 @@ protected static Task<bool> InternalAwaitConditionAsync(Func<bool> conditionIsFu
+        /// <param name="cancellationToken"></param>
         /// <returns>TBD</returns>
-        protected static bool InternalAwaitCondition(Func<bool> conditionIsFulfilled, TimeSpan max, TimeSpan? interval, Action<string, object[]> fail, ILoggingAdapter logger)
+        protected static bool InternalAwaitCondition(Func<bool> conditionIsFulfilled, TimeSpan max, TimeSpan? interval, Action<string, object[]> fail, ILoggingAdapter logger, CancellationToken cancellationToken = default)
         {
-            max.EnsureIsPositiveFinite("max");
-            var start = Now;
-            var stop = start + max;
-            ConditionalLog(logger, "Awaiting condition for {0}.{1}", max, interval.HasValue ? " Will sleep " + interval.Value + " between checks" : "");
-
-            while (!conditionIsFulfilled())
-            {
-                var now = Now;
-
-                if (now > stop)
-                {
-                    const string message = "Timeout {0} expired while waiting for condition.";
-                    ConditionalLog(logger, message, max);
-                    fail(message, new object[] { max });
-                    return false;
-                }
-                var sleepDuration = (stop - now).Min(interval);
-                Thread.Sleep(sleepDuration);
-            }
-            ConditionalLog(logger, "Condition fulfilled after {0}", Now-start);
-            return true;
+            var task = InternalAwaitConditionAsync(conditionIsFulfilled, max, interval, fail, logger, cancellationToken);
+            task.WaitAndUnwrapException();
+            return task.Result;
         }
         
-        /// <summary>
-        /// Async version of <see cref="InternalAwaitCondition(System.Func{bool},System.TimeSpan,System.Nullable{System.TimeSpan},System.Action{string,object[]})"/>
-        /// </summary>
-        protected static async Task<bool> InternalAwaitConditionAsync(Func<bool> conditionIsFulfilled, TimeSpan max, TimeSpan? interval, Action<string, object[]> fail, ILoggingAdapter logger)
+        /// <inheritdoc cref="InternalAwaitCondition(Func{bool}, TimeSpan, TimeSpan?, Action{string, object[]}, ILoggingAdapter, CancellationToken)"/>
+        protected static async Task<bool> InternalAwaitConditionAsync(Func<bool> conditionIsFulfilled, TimeSpan max, TimeSpan? interval, Action<string, object[]> fail, ILoggingAdapter logger, CancellationToken cancellationToken = default)
         {
             max.EnsureIsPositiveFinite("max");
             var start = Now;
             var stop = start + max;
             ConditionalLog(logger, "Awaiting condition for {0}.{1}", max, interval.HasValue ? " Will sleep " + interval.Value + " between checks" : "");
 
-            while (!conditionIsFulfilled())
+            while (!conditionIsFulfilled() && !cancellationToken.IsCancellationRequested)
```

Comment:
This is wrong, because the method will report condition fulfilled if CancellationToken is cancelled.